File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/GameChat_Request.java
package org.emulinker.kaillera.controller.v086.protocol;

import com.google.auto.value.AutoValue;
import org.emulinker.kaillera.controller.messaging.MessageFormatException;

@AutoValue
public abstract class GameChat_Request extends GameChat {
  private static final String DESC = "In-Game Chat Request";
  private static final String USERNAME = "";

  public static AutoValue_GameChat_Request create(int messageNumber, String message)
      throws MessageFormatException {
    V086Message.validateMessageNumber(messageNumber, DESC);

    return new AutoValue_GameChat_Request(messageNumber, ID, DESC, USERNAME, message);
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/GameChat_Request.kt
package org.emulinker.kaillera.controller.v086.protocol

import org.emulinker.kaillera.controller.messaging.MessageFormatException

private const val DESC = "In-Game Chat Request"

data class GameChat_Request
    @Throws(MessageFormatException::class)
    constructor(override val messageNumber: Int, override val message: String) : GameChat() {

  override val shortName = DESC
  override val messageId = ID
  override val username = ""

  init {
    validateMessageNumber(messageNumber, DESC)
  }
}
